src/disassembler.rs

```diff
@@ -383,6 +386,7 @@ pub fn to_insn_vec(prog: &[u8]) -> Vec<HLInsn> {
 /// neg64 r2
 /// exit
 /// ```
+#[cfg(feature = "std")]
 pub fn disassemble(prog: &[u8]) {
```

Comment:
I have updated it so that it calls log::info! when under no_std. I think it basically replicates the behaviour without disabling it. The idea is that in `no_std` one can define a custom logger that calls into the OS-specific printing functionality that can then print to a log file or the serial port like so:
```
impl Log for RiotLogger {
    fn enabled(&self, metadata: &Metadata<'_>) -> bool {
        metadata.level() <= self.level
    }

    fn log(&self, record: &Record<'_>) {
        if self.enabled(record.metadata()) {
            println!(
                "{}:{} -- {}",
                record.level(),
                record.target(),
                record.args()
            );
        }
    }

    fn flush(&self) {}
}
```
The example above is the logger that I used for RIOT applications.

Reply:
> The example above is the logger that I used for RIOT applications.

Just to double-check and be 100% clear here, this is some code that _you_ contributed to RIOT, not something in RIOT from which you took inspiration, right? I'm asking because I see RIOT is LGPL, so we can't reproduce its code in rbpf if you weren't the one to contribute it in the first place (although reading again, you mention RIOT _applications_ which is maybe something different).

Reply:
Yes that's right, this is just some custom application-specific code that I have in an app built on top of riot. Additionally, this snippet of code was just to illustrate to you how one might go about implementing the logger behaviour in their system. It's not actually contributed anywhere in rbpf. There is nothing in this PR that was taken / inspired by RIOT, I was just trying to motivate why using the log crate is a good idea here to allow users to specify where things should get logged to

Reply:
OK perfect, thank you for confirming :slightly_smiling_face: